version: "3"

services:
  cloudreve:
    container_name: cloudreve
    image: cloudreve/cloudreve:latest
    networks:
      - traefik
      - internal
    restart: unless-stopped
    expose:
      - "5212"
    volumes:
      - temp_data:/data
      - ./cloudreve/uploads:/cloudreve/uploads
      - ./cloudreve/conf.ini:/cloudreve/conf.ini
      - ./cloudreve/cloudreve.db:/cloudreve/cloudreve.db
      - ./cloudreve/avatar:/cloudreve/avatar
    depends_on:
      - aria2
    labels:
      - "traefik.enable=true"
      - "traefik.docker.network=traefik"
      - "traefik.http.routers.whoami.rule=Host(`${DOMAIN}`)"
      - "traefik.http.routers.whoami.entrypoints=websecure"
      - "traefik.http.routers.whoami.tls=true"
  aria2:
    container_name: aria2
    image: p3terx/aria2-pro
    networks:
      - internal
    restart: unless-stopped
    environment:
      - RPC_SECRET=${RPC_SECRET}
      - RPC_PORT=${RPC_PORT}
    volumes:
      - ./aria2/config:/config
      - temp_data:/data

volumes:
  temp_data:
    driver: local
    driver_opts:
      type: none
      device: $PWD/data
      o: bind

networks:
  traefik:
    external: true
  internal:
    external: false
    driver: bridge